scripts/upgrade.ps1

```
@@ -0,0 +1,983 @@
+    # Both rename endpoints must stay on the target volume. WorkRoot may be on
+    # a redirected TEMP volume, where File.Move cannot provide this swap.
+    $displaced=Join-Path $parent ("."+$name+".phase-one-displaced-"+[guid]::NewGuid().ToString("N"))
+    Write-SecuredRestoreCandidate -Source $Snapshot.Backup -Destination $candidate -FinalSddl $Snapshot.Sddl
+    $hadActive=Test-Path -LiteralPath $Snapshot.Active
+    if($hadActive){Assert-RealFile $Snapshot.Active "Rollback target";Set-PrivateFileAcl $Snapshot.Active;[IO.File]::Move($Snapshot.Active,$displaced)}
+    try{[IO.File]::Move($candidate,$Snapshot.Active)}catch{if($hadActive -and -not(Test-Path -LiteralPath $Snapshot.Active) -and (Test-Path -LiteralPath $displaced)){[IO.File]::Move($displaced,$Snapshot.Active)};throw}
+    if((Get-FileHash -LiteralPath $Snapshot.Active -Algorithm SHA256).Hash.ToLowerInvariant()-ne $Snapshot.Sha256 -or (Get-OwnerDaclSddl $Snapshot.Active)-ne $Snapshot.Sddl){Fail "Restored phase-one bytes or owner/DACL mismatch: $($Snapshot.Active)"}
+    if(Test-Path -LiteralPath $displaced){Set-PrivateFileAcl $displaced;try{Remove-Item -LiteralPath $displaced -Force -ErrorAction Stop}catch{Warn "Kept private rollback quarantine: $displaced"}}
+    Remove-PhaseOneQuarantines $Snapshot
+}
+function New-PhaseOneRollbackPlan {
```

PSScriptAnalyzer (1.25.0)

[warning] Function 'New-PhaseOneRollbackPlan' has verb that could change system state. Therefore, the function has to support 'ShouldProcess'.

(PSUseShouldProcessForStateChangingFunctions)